AISKU/snippet/snippet.js

```diff
@@ -1,106 +1,304 @@
-﻿var sdkInstance = "appInsightsSDK";
-window[sdkInstance] = "appInsights";
-var aiName = window[sdkInstance]; // provide non default instance name through key appInsightsSDK
-var aisdk = window[aiName] || (function (aiConfig) {
-    var appInsights = {
-        config: aiConfig
-    };
-    appInsights.initialize = true; // initialize sdk on download
-
-    // Assigning these to local variables allows them to be minified to save space:
-    var localDocument = document, localWindow = window, scriptText = "script";
-    setTimeout(function () {
-        var scriptElement = localDocument.createElement(scriptText);
-        scriptElement.src = aiConfig.url || "https://az416426.vo.msecnd.net/scripts/b/ai.2.min.js";
-        localDocument.getElementsByTagName(scriptText)[0].parentNode.appendChild(scriptElement);
-    });
-
-    // capture initial cookie
-    try {
-        appInsights.cookie = localDocument.cookie;
-    } catch (e) { }
-
-    appInsights.queue = [];
-    appInsights.version = 2.0;
-
-    function createLazyMethod(name) {
-        // Define a temporary method that queues-up a the real method call
-        appInsights[name] = function () {
-            // Capture the original arguments passed to the method
-            var originalArguments = arguments;
-            // Queue-up a call to the real method
-            appInsights.queue.push(function () {
-                // Invoke the real method with the captured original arguments
-                appInsights[name].apply(appInsights, originalArguments);
-            });
-        };
-    }
+﻿(function (win, doc, snipConfig) {
+    var locn = win.location;
+    var helpLink = "https://aka.ms/<TBD>";
+    var scriptText = "script";
+    var strInstrumentationKey = "instrumentationKey";
+    var strIngestionendpoint = "ingestionendpoint";
+    var strDisableExceptionTracking = "disableExceptionTracking";
+    var strAiDevice = "ai.device.";
+    var strAiOperationName = "ai.operation.name";
+    var strAiSdkVersion = "ai.internal.sdkVersion";
+    var strToLowerCase = "toLowerCase";
+    var strEmpty = "";
 
-    var method = ["Event", "PageView", "Exception", "Trace", "DependencyData", "Metric", "PageViewPerformance"];
-    while (method.length) {
-        createLazyMethod("track" + method.pop());
+    var strPostMethod = "POST";
+    var sdkInstanceName = "appInsightsSDK";         // required for Initialization to find the current instance
+    var aiName = snipConfig.name || "appInsights";  // provide non default instance name through snipConfig name value
+    if (snipConfig.name || win[sdkInstanceName]) {
+        // Only set if supplied or another name is defined to avoid polluting the global namespace
+        win[sdkInstanceName] = aiName;
     }
+    var aisdk = win[aiName] || (function (aiConfig) {
+        var loadFailed = false;
+        var handled = false;
+        var appInsights = {
+            initialize: true,   // initialize sdk on download
```

Comment:
Mostly no as this would break backward compatibility. Older scripts (and the current) require
queue - this is the documented way applications "wait" to process extra initialization until after the SDK is ready
initialize - without this the SDK wont call getAppInsights()
version - same as initialize, but also needs to be "2.0" (this is why I added snipVer so we can track the actual snippet version)
config - same again backward compatibility
snipVer - I could reduce to sv, would need to update other PR as well -- will consider

Reply:
Updated snipVer to sv